File: java/src/main/java/itp/itp1_6/ITP1_6_D.java
package itp.itp1_6;

import java.io.PrintWriter;
import java.util.Arrays;
import java.util.Scanner;

public class ITP1_6_D {

  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    PrintWriter pw = new PrintWriter(System.out);

    int n = sc.nextInt();
    int m = sc.nextInt();

    int[][] a = new int[n][m];
    for (int i = 0; i < n; i++) {
      for (int j = 0; j < m; j++) {
        a[i][j] = sc.nextInt();
      }
    }

    int[][] ans = new int[n][m];

    for (int i = 0; i < m; i++) {      
      int b = sc.nextInt();
      for (int j = 0; j < n; j++) {
        ans[j][i] = a[j][i] * b;
      }
    }

    for (int[] x : ans) {
      pw.println(Arrays.stream(x).sum());
    }

    pw.flush();
    sc.close();
  }
}

File: kotlin/src/itp/itp1_6/ITP1_6_D.kt
package itp.itp1_6

fun main(args: Array<String>) {
    val (n, m) = readLine()!!.split(" ").map(String::toInt)
    val A = (1..n).map { readLine()!!.split(" ").map(String::toInt) }
    val b = (1..m).map { readLine()!!.toInt() }

    (0 until n).forEach { ni ->
        println((0 until m).map { mi -> A[ni][mi] * b[mi] }.sum())
    }
}
